include/snitch/snitch_macros_reporter.hpp

```diff
@@ -5,13 +5,18 @@
 #include "snitch/snitch_macros_utility.hpp"
 #include "snitch/snitch_registry.hpp"
 
-#define SNITCH_REGISTER_REPORTER_CALLBACKS(NAME, ...)                                              \
-    static const std::string_view SNITCH_MACRO_CONCAT(reporter_id_, __COUNTER__)                   \
-        [[maybe_unused]] = snitch::tests.add_reporter(NAME, __VA_ARGS__)
+#if !(SNITCH_DISABLE)
+#    define SNITCH_REGISTER_REPORTER_CALLBACKS(NAME, ...)                                          \
+        static const std::string_view SNITCH_MACRO_CONCAT(reporter_id_, __COUNTER__)               \
+            [[maybe_unused]] = snitch::tests.add_reporter(NAME, __VA_ARGS__)
 
-#define SNITCH_REGISTER_REPORTER(NAME, TYPE)                                                       \
-    static const std::string_view SNITCH_MACRO_CONCAT(reporter_id_, __COUNTER__)                   \
-        [[maybe_unused]] = snitch::tests.add_reporter<TYPE>(NAME)
+#    define SNITCH_REGISTER_REPORTER(NAME, TYPE)                                                   \
+        static const std::string_view SNITCH_MACRO_CONCAT(reporter_id_, __COUNTER__)               \
+            [[maybe_unused]] = snitch::tests.add_reporter<TYPE>(NAME)
+#else // SNITCH_DISABLE
+#    define SNITCH_REGISTER_REPORTER_CALLBACKS(NAME, ...) /* nothing */
+#    define SNITCH_REGISTER_REPORTER(NAME, TYPE) static_assert(NAME)
```

Comment:
Any reason to use `static_assert()` here instead of `/* nothing */`? I'm not sure why the two definitions are different, I think they could be the same (i.e., either both use `static_assert()` or both use `/* nothing */`)?